HIP/hip-551.md

```diff
@@ -0,0 +1,89 @@
+---
+hip: 551
+title: Batch transactions
+author: Piotr Swierzy <piotr.swierzy@arianelabs.com>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Review
+created: 2022-07-25
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/531
+updated: 2022-09-19, 2022-10-04
+---
+
+## Abstract
+
+This HIP defines a mechanism to execute batch transactions such that a series of transactions(HAPI calls) depending on each other can be rolled into one transaction that passes the ACID test (atomicity, consistency, isolation, and durability)
+
+## Motivation
+
+The existing implementation of transactions in the Hedera network does not allow multiple different HAPI transactions to be called in one single network transaction that would have all the ACID properties.
+This makes it impossible to create more complicated flows without using smart contracts (which do not support all the HAPI transactions at this point) and listening to the mirror node to check the status of the previous transaction.
+This way we can also achieve an abstraction away from smart contracts
```

Comment:
This also means that if any of the contract calls at the end are reverted, then the approvals, associates, and account updates would be undone. This would improve the security of the network by reducing dangling allowances and auto association slots.